src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -382,11 +382,51 @@ ShouldSyncTableMetadata(Oid relationId)
 	CitusTableCacheEntry *tableEntry = GetCitusTableCacheEntry(relationId);
 
 	bool streamingReplicated =
-		(tableEntry->replicationModel == REPLICATION_MODEL_STREAMING);
+		tableEntry->replicationModel == REPLICATION_MODEL_STREAMING;
+	bool mxTable =
+		streamingReplicated && IsCitusTableTypeCacheEntry(tableEntry, HASH_DISTRIBUTED);
+	bool citusTableWithNoDistKey =
+		IsCitusTableTypeCacheEntry(tableEntry, CITUS_TABLE_WITH_NO_DIST_KEY);
+
+	return ShouldSyncTableMetadataInternal(mxTable, citusTableWithNoDistKey);
+}
+
+
+/*
+ * ShouldSyncTableMetadataViaCatalog checks if the metadata of a distributed table should
+ * be propagated to metadata workers, i.e. the table is an MX table or reference table.
+ * Tables with streaming replication model (which means RF=1) and hash distribution are
+ * considered as MX tables while tables with none distribution are reference tables.
+ *
+ * ShouldSyncTableMetadataViaCatalog does not use the CitusTableCache, and use only the
+ * catalog tables.
+ */
+bool
+ShouldSyncTableMetadataViaCatalog(Oid relationId)
+{
+	if (!OidIsValid(relationId) || !IsCitusTableViaCatalog(relationId))
+	{
+		return false;
+	}
+
+	char replicationModel = ReplicationModelViaCatalog(relationId);
```

Comment:
ShouldSyncTableMetadata will no longer look at replication model after https://github.com/citusdata/citus/pull/5392/files#

Reply:
I updated `ShouldSyncTableMetadata*` functions during a rebase after #5392 is merged to master. We do not care about the replication model now.